### PR TITLE
test: Wrap proto string in std::string before comparison

### PR DESCRIPTION
**Pull request by jamessynge**
Risk Level: Low
Testing: Unit
Docs Changes: N/A
Release Notes: N/A

Signed-off-by: James Synge <jamessynge@google.com>

### CODE REVIEW

File: test/integration/hds_integration_test.cc [hunk 1]
```diff
@@ -179,8 +179,9 @@ class HdsIntegrationTest : public HttpIntegrationTest,
     if (address->ip()->port() != endpoint.endpoint().address().socket_address().port_value()) {
       return false;
     }
+    // SocketAddress.address is a proto string, not std::string, so convert it before comparing.
     if (address->ip()->addressAsString() !=
-        endpoint.endpoint().address().socket_address().address()) {
+        std::string(endpoint.endpoint().address().socket_address().address())) {
```
Comment (jmarantz) [inline test/integration/hds_integration_test.cc]: You could also wrap both sides in absl::string_view, which would not involve any allocation or buffer-copies.

Comment (jmarantz) [reply test/integration/hds_integration_test.cc]: actually you can wrap just one side, as absl::string_view can compare itself with strings.

**Comment by jamessynge** [comment]
I looked at the log of the "ci/circleci: mac" failure, and it looked to be in setup of the build env, rather than building and testing the change. Could someone restart that build?

**Comment by htuch** [comment]
@jamessynge sorry, thought I had rerun it, slight change to CircleCI flow recently, hopefully should be done soon.

**Pull request merged**